src/core/simulation_stages.jl

```diff
@@ -134,12 +128,9 @@ function reset!(stage::Stage)
         @info("Stage $(stage.internal.number) will be reset by the build call")
     end
```

Comment:
This if block should no longer be needed.  You should be able to make it an `@assert`.

Reply:
I think it is still needed, but will update the message.